File: binding/include/charts/libqpieslice.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qpieslice.html
pub const qpieslice = struct {

    /// New constructs a new QPieSlice object.
    pub fn New() QtC.QPieSlice {
        return qtc.QPieSlice_new();
    }


    /// New2 constructs a new QPieSlice object.
    pub fn New2(label: []const u8, value: f64) QtC.QPieSlice {
        const label_str = qtc.libqt_string{
    .len = label.len,
    .data = label.ptr,
};
return qtc.QPieSlice_new2(label_str, value);
    }


    /// New3 constructs a new QPieSlice object.
    pub fn New3(parent: ?*anyopaque) QtC.QPieSlice {
        return qtc.QPieSlice_new3(@ptrCast(parent));
    }


    /// New4 constructs a new QPieSlice object.
    pub fn New4(label: []const u8, value: f64, parent: ?*anyopaque) QtC.QPieSlice {
        const label_str = qtc.libqt_string{
    .len = label.len,
    .data = label.ptr,
};
return qtc.QPieSlice_new4(label_str, value, @ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPieSlice_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpieslice.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetLabel(self: ?*anyopaque, label: []const u8) void {
        const label_str = qtc.libqt_string{
    .len = label.len,
    .data = label.ptr,
};
qtc.QPieSlice_SetLabel(@ptrCast(self), label_str);
    }

    pub fn Label(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPieSlice_Label(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpieslice.Label: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetValue(self: ?*anyopaque, value: f64) void {
        qtc.QPieSlice_SetValue(@ptrCast(self), value);
    }

    pub fn Value(self: ?*anyopaque, ) f64 {
        return qtc.QPieSlice_Value(@ptrCast(self));
    }

    pub fn SetLabelVisible(self: ?*anyopaque, ) void {
        qtc.QPieSlice_SetLabelVisible(@ptrCast(self));
    }

    pub fn IsLabelVisible(self: ?*anyopaque, ) bool {
        return qtc.QPieSlice_IsLabelVisible(@ptrCast(self));
    }

    pub fn LabelPosition(self: ?*anyopaque, ) i32 {
        return qtc.QPieSlice_LabelPosition(@ptrCast(self));
    }

    pub fn SetLabelPosition(self: ?*anyopaque, position: i32) void {
        qtc.QPieSlice_SetLabelPosition(@ptrCast(self), @intCast(position));
    }

    pub fn SetExploded(self: ?*anyopaque, ) void {
        qtc.QPieSlice_SetExploded(@ptrCast(self));
    }

    pub fn IsExploded(self: ?*anyopaque, ) bool {
        return qtc.QPieSlice_IsExploded(@ptrCast(self));
    }

    pub fn SetPen(self: ?*anyopaque, pen: ?*anyopaque) void {
        qtc.QPieSlice_SetPen(@ptrCast(self), @ptrCast(pen));
    }

    pub fn Pen(self: ?*anyopaque, ) QtC.QPen {
        return qtc.QPieSlice_Pen(@ptrCast(self));
    }

    pub fn BorderColor(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QPieSlice_BorderColor(@ptrCast(self));
    }

    pub fn SetBorderColor(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QPieSlice_SetBorderColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn BorderWidth(self: ?*anyopaque, ) i32 {
        return qtc.QPieSlice_BorderWidth(@ptrCast(self));
    }

    pub fn SetBorderWidth(self: ?*anyopaque, width: i32) void {
        qtc.QPieSlice_SetBorderWidth(@ptrCast(self), width);
    }

    pub fn SetBrush(self: ?*anyopaque, brush: ?*anyopaque) void {
        qtc.QPieSlice_SetBrush(@ptrCast(self), @ptrCast(brush));
    }

    pub fn Brush(self: ?*anyopaque, ) QtC.QBrush {
        return qtc.QPieSlice_Brush(@ptrCast(self));
    }

    pub fn Color(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QPieSlice_Color(@ptrCast(self));
    }

    pub fn SetColor(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QPieSlice_SetColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn SetLabelBrush(self: ?*anyopaque, brush: ?*anyopaque) void {
        qtc.QPieSlice_SetLabelBrush(@ptrCast(self), @ptrCast(brush));
    }

    pub fn LabelBrush(self: ?*anyopaque, ) QtC.QBrush {
        return qtc.QPieSlice_LabelBrush(@ptrCast(self));
    }

    pub fn LabelColor(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QPieSlice_LabelColor(@ptrCast(self));
    }

    pub fn SetLabelColor(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QPieSlice_SetLabelColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn SetLabelFont(self: ?*anyopaque, font: ?*anyopaque) void {
        qtc.QPieSlice_SetLabelFont(@ptrCast(self), @ptrCast(font));
    }

    pub fn LabelFont(self: ?*anyopaque, ) QtC.QFont {
        return qtc.QPieSlice_LabelFont(@ptrCast(self));
    }

    pub fn SetLabelArmLengthFactor(self: ?*anyopaque, factor: f64) void {
        qtc.QPieSlice_SetLabelArmLengthFactor(@ptrCast(self), factor);
    }

    pub fn LabelArmLengthFactor(self: ?*anyopaque, ) f64 {
        return qtc.QPieSlice_LabelArmLengthFactor(@ptrCast(self));
    }

    pub fn SetExplodeDistanceFactor(self: ?*anyopaque, factor: f64) void {
        qtc.QPieSlice_SetExplodeDistanceFactor(@ptrCast(self), factor);
    }

    pub fn ExplodeDistanceFactor(self: ?*anyopaque, ) f64 {
        return qtc.QPieSlice_ExplodeDistanceFactor(@ptrCast(self));
    }

    pub fn Percentage(self: ?*anyopaque, ) f64 {
        return qtc.QPieSlice_Percentage(@ptrCast(self));
    }

    pub fn StartAngle(self: ?*anyopaque, ) f64 {
        return qtc.QPieSlice_StartAngle(@ptrCast(self));
    }

    pub fn AngleSpan(self: ?*anyopaque, ) f64 {
        return qtc.QPieSlice_AngleSpan(@ptrCast(self));
    }

    pub fn Series(self: ?*anyopaque, ) QtC.QPieSeries {
        return qtc.QPieSlice_Series(@ptrCast(self));
    }

    pub fn Clicked(self: ?*anyopaque, ) void {
        qtc.QPieSlice_Clicked(@ptrCast(self));
    }

    pub fn OnClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_Clicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Hovered(self: ?*anyopaque, state: bool) void {
        qtc.QPieSlice_Hovered(@ptrCast(self), state);
    }

    pub fn OnHovered(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QPieSlice_Connect_Hovered(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Pressed(self: ?*anyopaque, ) void {
        qtc.QPieSlice_Pressed(@ptrCast(self));
    }

    pub fn OnPressed(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_Pressed(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Released(self: ?*anyopaque, ) void {
        qtc.QPieSlice_Released(@ptrCast(self));
    }

    pub fn OnReleased(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_Released(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn DoubleClicked(self: ?*anyopaque, ) void {
        qtc.QPieSlice_DoubleClicked(@ptrCast(self));
    }

    pub fn OnDoubleClicked(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_DoubleClicked(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LabelChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_LabelChanged(@ptrCast(self));
    }

    pub fn OnLabelChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_LabelChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ValueChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_ValueChanged(@ptrCast(self));
    }

    pub fn OnValueChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_ValueChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LabelVisibleChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_LabelVisibleChanged(@ptrCast(self));
    }

    pub fn OnLabelVisibleChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_LabelVisibleChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PenChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_PenChanged(@ptrCast(self));
    }

    pub fn OnPenChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_PenChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn BrushChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_BrushChanged(@ptrCast(self));
    }

    pub fn OnBrushChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_BrushChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LabelBrushChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_LabelBrushChanged(@ptrCast(self));
    }

    pub fn OnLabelBrushChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_LabelBrushChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LabelFontChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_LabelFontChanged(@ptrCast(self));
    }

    pub fn OnLabelFontChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_LabelFontChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PercentageChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_PercentageChanged(@ptrCast(self));
    }

    pub fn OnPercentageChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_PercentageChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn StartAngleChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_StartAngleChanged(@ptrCast(self));
    }

    pub fn OnStartAngleChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_StartAngleChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn AngleSpanChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_AngleSpanChanged(@ptrCast(self));
    }

    pub fn OnAngleSpanChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_AngleSpanChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ColorChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_ColorChanged(@ptrCast(self));
    }

    pub fn OnColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_ColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn BorderColorChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_BorderColorChanged(@ptrCast(self));
    }

    pub fn OnBorderColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_BorderColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn BorderWidthChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_BorderWidthChanged(@ptrCast(self));
    }

    pub fn OnBorderWidthChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_BorderWidthChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LabelColorChanged(self: ?*anyopaque, ) void {
        qtc.QPieSlice_LabelColorChanged(@ptrCast(self));
    }

    pub fn OnLabelColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QPieSlice_Connect_LabelColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPieSlice_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpieslice.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QPieSlice_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qpieslice.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetLabelVisible1(self: ?*anyopaque, visible: bool) void {
        qtc.QPieSlice_SetLabelVisible1(@ptrCast(self), visible);
    }

    pub fn SetExploded1(self: ?*anyopaque, exploded: bool) void {
        qtc.QPieSlice_SetExploded1(@ptrCast(self), exploded);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QPieSlice_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/pieslice.html#types
pub const enums = struct {
    pub const LabelPosition = enum {
        pub const LabelOutside: i32 = 0;
        pub const LabelInsideHorizontal: i32 = 1;
        pub const LabelInsideTangential: i32 = 2;
        pub const LabelInsideNormal: i32 = 3;
    };

};

File: binding/src/charts/libqpieslice.cpp
#include <QBrush>
#include <QColor>
#include <QFont>
#include <QObject>
#include <QPen>
#include <QPieSeries>
#include <QPieSlice>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qpieslice.h>
#include "libqpieslice.h"
#include "libqpieslice.hxx"

QPieSlice* QPieSlice_new() {
	 return new QPieSlice();
}

QPieSlice* QPieSlice_new2(libqt_string label, double value) {
	 return new QPieSlice(QString::fromUtf8(label.data, label.len), value);
}

QPieSlice* QPieSlice_new3(QObject* parent) {
	 return new QPieSlice(parent);
}

QPieSlice* QPieSlice_new4(libqt_string label, double value, QObject* parent) {
	 return new QPieSlice(QString::fromUtf8(label.data, label.len), value, parent);
}

libqt_string QPieSlice_Tr(const char* s) {
	QString _ret = QPieSlice::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QPieSlice_SetLabel(QPieSlice* self, libqt_string label) {
	self->setLabel(QString::fromUtf8(label.data, label.len));
}

libqt_string QPieSlice_Label(const QPieSlice* self) {
	QString _ret = self->label();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QPieSlice_SetValue(QPieSlice* self, double value) {
	self->setValue(value);
}

double QPieSlice_Value(const QPieSlice* self) {
	return self->value();
}

void QPieSlice_SetLabelVisible(QPieSlice* self) {
	self->setLabelVisible();
}

bool QPieSlice_IsLabelVisible(const QPieSlice* self) {
	return self->isLabelVisible();
}

int QPieSlice_LabelPosition(QPieSlice* self) {
	return self->labelPosition();
}

void QPieSlice_SetLabelPosition(QPieSlice* self, int position) {
	self->setLabelPosition(static_cast<QPieSlice::LabelPosition>(position));
}

void QPieSlice_SetExploded(QPieSlice* self) {
	self->setExploded();
}

bool QPieSlice_IsExploded(const QPieSlice* self) {
	return self->isExploded();
}

void QPieSlice_SetPen(QPieSlice* self, const QPen* pen) {
	self->setPen(*pen);
}

QPen* QPieSlice_Pen(const QPieSlice* self) {
	return new QPen(self->pen());
}

QColor* QPieSlice_BorderColor(QPieSlice* self) {
	return new QColor(self->borderColor());
}

void QPieSlice_SetBorderColor(QPieSlice* self, QColor* color) {
	self->setBorderColor(*color);
}

int QPieSlice_BorderWidth(QPieSlice* self) {
	return self->borderWidth();
}

void QPieSlice_SetBorderWidth(QPieSlice* self, int width) {
	self->setBorderWidth(width);
}

void QPieSlice_SetBrush(QPieSlice* self, const QBrush* brush) {
	self->setBrush(*brush);
}

QBrush* QPieSlice_Brush(const QPieSlice* self) {
	return new QBrush(self->brush());
}

QColor* QPieSlice_Color(QPieSlice* self) {
	return new QColor(self->color());
}

void QPieSlice_SetColor(QPieSlice* self, QColor* color) {
	self->setColor(*color);
}

void QPieSlice_SetLabelBrush(QPieSlice* self, const QBrush* brush) {
	self->setLabelBrush(*brush);
}

QBrush* QPieSlice_LabelBrush(const QPieSlice* self) {
	return new QBrush(self->labelBrush());
}

QColor* QPieSlice_LabelColor(QPieSlice* self) {
	return new QColor(self->labelColor());
}

void QPieSlice_SetLabelColor(QPieSlice* self, QColor* color) {
	self->setLabelColor(*color);
}

void QPieSlice_SetLabelFont(QPieSlice* self, const QFont* font) {
	self->setLabelFont(*font);
}

QFont* QPieSlice_LabelFont(const QPieSlice* self) {
	return new QFont(self->labelFont());
}

void QPieSlice_SetLabelArmLengthFactor(QPieSlice* self, double factor) {
	self->setLabelArmLengthFactor(factor);
}

double QPieSlice_LabelArmLengthFactor(const QPieSlice* self) {
	return self->labelArmLengthFactor();
}

void QPieSlice_SetExplodeDistanceFactor(QPieSlice* self, double factor) {
	self->setExplodeDistanceFactor(factor);
}

double QPieSlice_ExplodeDistanceFactor(const QPieSlice* self) {
	return self->explodeDistanceFactor();
}

double QPieSlice_Percentage(const QPieSlice* self) {
	return self->percentage();
}

double QPieSlice_StartAngle(const QPieSlice* self) {
	return self->startAngle();
}

double QPieSlice_AngleSpan(const QPieSlice* self) {
	return self->angleSpan();
}

QPieSeries* QPieSlice_Series(const QPieSlice* self) {
	return self->series();
}

void QPieSlice_Clicked(QPieSlice* self) {
	self->clicked();
}

void QPieSlice_Connect_Clicked(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::clicked, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_Hovered(QPieSlice* self, bool state) {
	self->hovered(state);
}

void QPieSlice_Connect_Hovered(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*, bool) = reinterpret_cast<void (*)(QPieSlice*, bool)>(slot);
    QPieSlice::connect(self, &QPieSlice::hovered, [self, slotFunc](bool state) {
	slotFunc(self, state);
    });
}

void QPieSlice_Pressed(QPieSlice* self) {
	self->pressed();
}

void QPieSlice_Connect_Pressed(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::pressed, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_Released(QPieSlice* self) {
	self->released();
}

void QPieSlice_Connect_Released(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::released, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_DoubleClicked(QPieSlice* self) {
	self->doubleClicked();
}

void QPieSlice_Connect_DoubleClicked(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::doubleClicked, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_LabelChanged(QPieSlice* self) {
	self->labelChanged();
}

void QPieSlice_Connect_LabelChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::labelChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_ValueChanged(QPieSlice* self) {
	self->valueChanged();
}

void QPieSlice_Connect_ValueChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::valueChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_LabelVisibleChanged(QPieSlice* self) {
	self->labelVisibleChanged();
}

void QPieSlice_Connect_LabelVisibleChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::labelVisibleChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_PenChanged(QPieSlice* self) {
	self->penChanged();
}

void QPieSlice_Connect_PenChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::penChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_BrushChanged(QPieSlice* self) {
	self->brushChanged();
}

void QPieSlice_Connect_BrushChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::brushChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_LabelBrushChanged(QPieSlice* self) {
	self->labelBrushChanged();
}

void QPieSlice_Connect_LabelBrushChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::labelBrushChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_LabelFontChanged(QPieSlice* self) {
	self->labelFontChanged();
}

void QPieSlice_Connect_LabelFontChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::labelFontChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_PercentageChanged(QPieSlice* self) {
	self->percentageChanged();
}

void QPieSlice_Connect_PercentageChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::percentageChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_StartAngleChanged(QPieSlice* self) {
	self->startAngleChanged();
}

void QPieSlice_Connect_StartAngleChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::startAngleChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_AngleSpanChanged(QPieSlice* self) {
	self->angleSpanChanged();
}

void QPieSlice_Connect_AngleSpanChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::angleSpanChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_ColorChanged(QPieSlice* self) {
	self->colorChanged();
}

void QPieSlice_Connect_ColorChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::colorChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_BorderColorChanged(QPieSlice* self) {
	self->borderColorChanged();
}

void QPieSlice_Connect_BorderColorChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::borderColorChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_BorderWidthChanged(QPieSlice* self) {
	self->borderWidthChanged();
}

void QPieSlice_Connect_BorderWidthChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::borderWidthChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

void QPieSlice_LabelColorChanged(QPieSlice* self) {
	self->labelColorChanged();
}

void QPieSlice_Connect_LabelColorChanged(QPieSlice* self, intptr_t slot) {
    void (*slotFunc)(QPieSlice*) = reinterpret_cast<void (*)(QPieSlice*)>(slot);
    QPieSlice::connect(self, &QPieSlice::labelColorChanged, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QPieSlice_Tr2(const char* s, const char* c) {
	QString _ret = QPieSlice::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QPieSlice_Tr3(const char* s, const char* c, int n) {
	QString _ret = QPieSlice::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QPieSlice_SetLabelVisible1(QPieSlice* self, bool visible) {
	self->setLabelVisible(visible);
}

void QPieSlice_SetExploded1(QPieSlice* self, bool exploded) {
	self->setExploded(exploded);
}

void QPieSlice_Delete(QPieSlice* self) {
    delete self;
}

